automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java

```diff
@@ -73,13 +74,20 @@ public void sampleOccurred(SampleEvent sampleEvent) {
                                             result.getResponseMessage().replaceAll("\"", "\\\""),
                                             "\", \"Sampler Data\": \"",
                                             samplerData.replaceAll("\"", "\\\""), "\"}");
-            // Persist result to the database
-            testCaseUOW.persistTestCase(result.getSampleLabel(), testScenario, result.isSuccessful(), failureMessage);
+            // Add result to concurrent list
+            TestCase testCase = new TestCase();
+            testCase.setName(result.getSampleLabel());
+            testCase.setTestScenario(testScenario);
+            testCase.setSuccess(result.isSuccessful());
+            testCase.setFailureMessage(failureMessage);
+            testCases.add(testCase);
             jmeterResultLogger.info(StringUtil.concatStrings("Test case :", result.getSampleLabel(),
                     " failed for scenario: ", testScenario.getName(), "\n", failureMessage, "\", \"Sampler Data\": \"",
                     result.getSamplerData().replaceAll("\"", "\\\""), "\"}"));
-        } catch (TestGridDAOException e) {
-            throw new RuntimeException(StringUtil.concatStrings("Error occurred when persisting test case."), e);
-        }
+
+    }
+
+    public List<TestCase> getTestCases() {
```

Comment:
This returns the list of testcases that has been executed, right? I mean, it won't list unless the test case has come into the `sampleOccurred` method, isn't it? I think we should document that.

Further, we need another boolean flag that specify whether all the test cases of a given scenario are ran or not. We don't need to do this part in this PR. Please raise a separate ticket.

Reply:
> This returns the list of testcases that has been executed, right? I mean, it won't list unless the test case has come into the sampleOccurred method, isn't it?

Correct.
> Please raise a separate ticket.

Raised #467

Reply:
Added a javadoc comment explaining the method in e59c4e3